src/main/java/reposense/parser/LocalRepoLocationParser.java

```diff
@@ -0,0 +1,88 @@
+     * branches specified.
+     */
+    public static List<String> parseLocalRepoLocation(String locationAndBranches, String defaultSpecifiedBranch) {
+        assert !isValidUrl(locationAndBranches);
+
+        if (isUsingDelimiter(locationAndBranches)) {
+            return parseLocalWithDelimiter(locationAndBranches);
+        } else {
+            return parseLocal(locationAndBranches, defaultSpecifiedBranch);
+        }
+    }
+
+    /**
+     * Parses a local repository path into a list containing location and branch information.
+     * The first item in the list is the location, while subsequent items are the
+     * branches specified.
+     * <br><br>
+     * The path to the repo can contain several ~ characters as well. The last ~ character will
+     * be used to parse the path into the array.
+     * <br><br>
+     * Example: home~Desktop~RepoSense~master is parsed to give { home~Desktop~RepoSense, master }
+     */
+    private static List<String> parseLocalWithDelimiter(String locationAndBranches) {
+        List<String> parsedLocationAndBranches = new ArrayList<>();
+        int lastLocationBranchesDelimiterIndex = locationAndBranches.lastIndexOf(
+                LOCATION_BRANCHES_DELIMITER);
+
+        String location = locationAndBranches.substring(0, lastLocationBranchesDelimiterIndex);
+        parsedLocationAndBranches.add(location);
+
+        String[] branches = locationAndBranches
+                .substring(lastLocationBranchesDelimiterIndex + 1).split(SPLIT_SAFE_BRANCHES_DELIMITER);
+        for (String branch : branches) {
+            parsedLocationAndBranches.add(branch);
+        }
+
+        if (branches.length == 0) { // by default, an empty branch is used
+            parsedLocationAndBranches.add("");
+        }
+
+        return parsedLocationAndBranches;
+    }
+
+    /**
+     * Parses a local repository path that is not using any additional special syntax
+     * into a 2-element list containing location and branch information.
+     * The first item in the list is the location, while the second item is the default branch
+     * from {@code author-config.csv} branch column.
+     */
+    private static List<String> parseLocal(String location, String defaultSpecifiedBranch) {
+        List<String> parsedLocationAndBranch = new ArrayList<>();
+        parsedLocationAndBranch.add(location);
+        parsedLocationAndBranch.add(defaultSpecifiedBranch);
+        return parsedLocationAndBranch;
+    }
+
+    /**
+     * Returns true if the given {@code locationAndBranch} is using the delimiter syntax.
+     * If the given {@code locationAndBranch} exists, then it is considered to not be
+     * using the delimiter syntax.
+     */
+    private static boolean isUsingDelimiter(String locationAndBranch) {
+        if (pathExists(locationAndBranch)) {
+            return false;
+        }
+
+        return locationAndBranch.lastIndexOf(LOCATION_BRANCHES_DELIMITER) != -1;
+    }
+}
```

Comment:
I think a more clear separation between the two types of parser (remote vs local) would be 
1) domain specific branch URL (e.g. the github branch url)
2) multiple remote/local repo possibly with branch specification

And you can probably handle the checking like this: Check if the path exists (doesn't matter if local or remote), if so then its standalone (neither 1 or 2). Otherwise, if `|` or `;` detected, it's case 2. Otherwise try to fit the specified location into 1). If it fails, then treat the argument as a standalone repo location. 

This idea somewhat assumes that we don't need to use the `~` separator

Reply:
Yes I believe this will be far more ideal. I was initially too focused on separating the local and remote repos rather than whether it is necessary, but an agnostic implementation would work just fine and better in terms of justifying the code.